awsbestpractices/deprecated-methods/DynamoDBRppApprovalDao.java

```diff
@@ -0,0 +1,124 @@
+import amazon.pricing.core.batching.BatchResult;
+import amazon.pricing.core.batching.BatchResultBuilder;
+import amazon.pricing.model.retailsystems.AmazonDomain;
+import amazon.pricing.model.retailsystems.DataCenterRegion;
+
+import com.amazon.pricing.ddb.partition.manager.tools.TableNameGenerator;
+import com.amazon.pricing.dynamodb.AttributeValues;
+import com.amazon.pricing.dynamodb.client.DynamoDBRetryableClient;
+import com.amazon.pricing.dynamodb.codec.DynamoDBCodec;
+import com.amazon.pricing.dynamodb.tablename.DomainAndRealmAwareTableName;
+import com.amazon.pricing.model.OfferListingKey;
+import com.amazon.pricing.retailpricepublisher.approval.codec.OfferListingKeyCodec;
+import com.amazon.pricing.retailpricepublisher.approval.model.ProposedPrice;
+import com.amazon.pricing.retailpricepublisher.exception.ApprovalException;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.model.AttributeAction;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.AttributeValueUpdate;
+import com.amazonaws.services.dynamodbv2.model.ComparisonOperator;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.amazonaws.services.dynamodbv2.model.ConditionalCheckFailedException;
+import com.amazonaws.services.dynamodbv2.model.DeleteItemRequest;
+import com.amazonaws.services.dynamodbv2.model.ExpectedAttributeValue;
+import com.amazonaws.services.dynamodbv2.model.PutItemRequest;
+import com.amazonaws.services.dynamodbv2.model.QueryRequest;
+import com.amazonaws.services.dynamodbv2.model.QueryResult;
+import com.amazonaws.services.dynamodbv2.model.UpdateItemRequest;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Iterables;
+
+public class DynamoDBRppApprovalDao implements RppApprovalDao {
+
+    private static Logger logger = LoggerFactory.getLogger(DynamoDBRppApprovalDao.class);
+
+    private final AmazonDynamoDB dynamoDBClient;
+
+    public DynamoDBRppApprovalDao(
+            final AmazonDynamoDB dynamoDBClient) {
+        this.dynamoDBClient = checkNotNull(dynamoDBClient);
+    }
+
+    private PutItemRequest firstChainedCall(final ProposedPrice proposedPrice) {
+        // Condition for not replacing existing record.
+        final Map<String, ExpectedAttributeValue> expectedAttributeValuesByName = newHashMap();
+        final ExpectedAttributeValue nullExpect = new ExpectedAttributeValue().withComparisonOperator(ComparisonOperator.NULL);
+        expectedAttributeValuesByName.put(OFFER_LISTING_KEY_ATTRIBUTE, nullExpect);
+        expectedAttributeValuesByName.put(VERSION_ATTRIBUTE, nullExpect);
+
+        final PutItemRequest request = new PutItemRequest()
+                .withExpected(expectedAttributeValuesByName)
+                .withTableName(proposedPricesTableName.getTableName())
+                .withItem(this.proposedPriceCodec.serialize(proposedPrice));
+        return request;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 64:73, 78:87.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.